src/services/arweave/AntContract.tsx

```diff
@@ -0,0 +1,299 @@
+import {
+  ANTContractState,
+  ArweaveDataProvider,
+  ArweaveTransactionID,
+  SmartweaveDataProvider,
+} from '../../types';
+
+export class AntContract {
+  private _provider: SmartweaveDataProvider & ArweaveDataProvider;
+  state?: ANTContractState = undefined;
+  antId;
+  walletAddress: ArweaveTransactionID;
+  newAntState: ANTContractState;
+  /**
+   * AntContractProvider
+   * @param walletAddress - @type {ArweaveTransactionID} - connected wallet address
+   * @param provider -  @type {ArweaveCompositeDataProvider}
+   * @param antId - @type {ArweaveTransactionID | undefined} -
+   * ANT ID to use for loading a pre-existing ANT's contract state. If left undefined, default new ANT state is loaded
+   */
+  constructor(
+    walletAddress: ArweaveTransactionID,
+    provider: SmartweaveDataProvider & ArweaveDataProvider,
+    antId?: ArweaveTransactionID,
+  ) {
+    this.walletAddress = walletAddress;
+    this.antId = antId;
+    this._provider = provider;
+    this.newAntState = {
```

Comment:
nit: can we use just `state` and updates happen there. i don't see the need for two.

Reply:
newAntState is a constant that is built using the walletAddress the class is given - changes dont happen there. Maybe DEFAULT_STATE would be a better name.

Reply:
can't we just set `this.state` to the default state before it loads? do we need two?